### ISSUE TITLE
Android doesn't show Google popup, always give Authorisation not granted

### ISSUE THREAD

**Issue by FirdousNath** [cph-cachet/flutter-plugins#383]
Latest 3.0.6 version with null-safety is broken on android. Please help me with a workaround.

**Comment by EunyoungY** [comment]
same happens to me

**Comment by FirdousNath** [comment]
as a work around we can use 3.0.4. But than i need to build without sound null safety

**Comment by vanessabin** [comment]
> as a work around we can use 3.0.4. But than i need to build without sound null safety

how did you make it run in 3.0.4? I even can't start the service on health 3.0.4

**Comment by Crewlan** [comment]
Hi @FirdousNath , @EunyoungY  and @vanessabin ,  i'm opened this PR #394  to solution this problem.

**Comment by FirdousNath** [comment]
> Hi @FirdousNath , @EunyoungY and @vanessabin , i'm opened this PR #394 to solution this problem.

how can I point to your PR in my pubspec.yaml ?

**Comment by Crewlan** [comment]
```
health: 
    git:
      url: git@github.com:Crewlan/flutter-plugins.git
      path: packages/health
      ref: master
```

Send your email to added in fork

**Comment by FirdousNath** [comment]
> > > Hi @FirdousNath , @EunyoungY and @vanessabin , i'm opened this PR #394 to solution this problem.
> > 
> > 
> > how can I point to your PR in my pubspec.yaml ?
> 
> ```
> health: 
>     git:
>       url: git@github.com:Crewlan/flutter-plugins.git
>       path: packages/health
>       ref: master
> ```
> 
> Send your email to added in fork

firunath@gmail.com

**Comment by FirdousNath** [comment]
its working with below code
```
`health: 
      git:
        url: https://github.com/Crewlan/flutter-plugins.git
        path: packages/health
        ref: master`
```

**Comment by Bisensortech** [comment]
its working Authorization ,  but it doesn't read steps data.

**Comment by skenderl** [comment]
> its working Authorization , but it doesn't read steps data.

 @Bisensortech Looks like it's the same issue in #332
> So apparently the **activityRecognition**-Permission was introduced starting with Android 10. Google fit requires this permission for step count specifically. https://developer.android.com/about/versions/10/privacy/changes#physical-activity-recognition
> 
> I feel like this should be mentioned in the health README

>also don't forget to add permission in AndroidManifest.xml
> `<uses-permission android:name="android.permission.ACTIVITY_RECOGNITION" />`

**Comment by bardram** [comment]
Fixed with PR #394  and released as version 3.1.1
